internal/store/engine_state.go

```diff
@@ -65,6 +65,9 @@ type EngineState struct {
 
 	TriggerMode  model.TriggerMode
 	TriggerQueue []model.ManifestName
+
+	Profiling   bool
+	ProfileFile *os.File
```

Comment:
it seems like a bad idea to have a pointer to an external OS thing in the engineState, especially as we start cloning/serializing/copying this state to the web ui, protobuf, etc

Reply:
Oh, I didn't realize that serializability was a requirement for EngineState, and I didn't really see anywhere else Upper stores state. I'll inject something into Upper to hold the state.